src/katsdpcontroller/product_controller.py

```diff
@@ -1497,6 +1497,28 @@ async def beam_delays(self, stream_name: str, coefficient_sets: Sequence[str]) -
             timeout=DELAYS_TIMEOUT,
         )
 
+    async def dsim_signals(self, dsim: str, signals_str: str, period: Optional[int] = None) -> int:
+        if self.state not in {ProductState.CAPTURING, ProductState.IDLE}:
+            raise FailReply(f"Cannot set dsim signals in state {self.state}")
+        for task in self.physical_graph:
+            if (
+                isinstance(task, tasks.ProductPhysicalTaskMixin)
+                and task.logical_node.task_type == "sim"
+                and task.name == dsim
+            ):
+                dsim_task = task
+                break
+        else:
+            raise FailReply(f"No dsim named {dsim}")
```

Comment:
Trying to compare notes and files my side, can't quite get to the bottom of my own question.
* Any particular reason you can't do the usual `stream = self._find_stream(...)`?
* I gather it's something along the lines of not needing to issue a `_multi_request` because it's direct to the dsim?

Reply:
This request is a bit awkward because there isn't a one-to-one mapping between streams and dsims. generator.py goes and finds two streams (e.g. m000h and m000v) which have the same antenna and sample rate, and runs them within one dsim.